kratos/utilities/auxiliar_model_part_utilities.cpp

```diff
@@ -372,4 +373,294 @@ void AuxiliarModelPartUtilities::RemoveConditionsAndBelongingsFromAllLevels(Flag
     aux_utility.RemoveConditionsAndBelongings(IdentifierFlag);
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+ModelPart& AuxiliarModelPartUtilities::DeepCopyModelPart(
+    const std::string& rNewModelPartName,
+    Model* pModel
+    )
+{
+    // First we check if the model is not null
+    if (pModel == nullptr)
+        pModel = &mrModelPart.GetModel();
+
+    // We create the new model part
+    ModelPart& r_model_part = pModel->CreateModelPart(rNewModelPartName);
+
+    /// We will copy the member variables of the model part one by one
+
+    // We copy the buffer size (direct copy)
+    r_model_part.SetBufferSize(mrModelPart.GetBufferSize());
+
+    // We copy the process info
+    r_model_part.GetProcessInfo() = mrModelPart.GetProcessInfo();
+
+    // We copy the list of variables (using copy constructor)
+    r_model_part.SetNodalSolutionStepVariablesList(Kratos::make_intrusive<VariablesList>(mrModelPart.GetNodalSolutionStepVariablesList()));
```

Comment:
To my understanding you are copying the intrusive pointer and not the list itself.

Reply:
This is creating a new list from scratch, because of that I am using the intrusive pointer constructor (make intrusive). Everything is deep copied, and therefore nothing is pointing to the original objects